### PR TITLE
feat: extend description mode with list and instruction properties

### PR DESCRIPTION
**Pull request by jonasbrunvoll**
### Background
Currently, the description modal display information through a  single string. This can look a bit strange, especially when the string starts to get long, and if it is desired to list a lot of information. 

This PR extends the description model with more properties to present information in a nicer way.  

#### Illustrations
<details>
<summary>Before</summary>
Skjermbilde 2024-11-11 kl  13 28 54

</details>

<details>
<summary>After</summary>
Skjermbilde 2024-11-11 kl  09 42 00

</details>

### Proposed solution
Add bullet point list and instruction properties to description modal.

### CODE REVIEW

File: src/page-modules/contact/components/form/form.module.css [hunk 1]
```diff
@@ -320,18 +321,25 @@
...
-  position: relative;
 }
 
 .modal_content > * + * {
-  margin-top: var(--spacings-large);
+  margin-top: var(--spacings-small);
 }
-
-.modal_header {
+* .modal_header {
```
Comment (adriansberg) [inline src/page-modules/contact/components/form/form.module.css:329]: Is the wildcard necessary here; could it be `.modal_content .modal_header`?

File: src/page-modules/contact/components/form/description-modal.tsx [hunk 2]
```diff
@@ -5,18 +5,22 @@ import { FocusScope } from '@react-aria/focus';
 import { MonoIcon } from '@atb/components/icon';
 import { useEffect, useRef } from 'react';
 import { Button } from '@atb/components/button';
-import { PageText, useTranslation } from '@atb/translations';
+import { PageText, TranslatedString, useTranslation } from '@atb/translations';
 
 type DescriptionModalProps = {
   title: string;
-  description: string;
+  modalDescription?: string;
```
Comment (adriansberg) [inline src/page-modules/contact/components/form/description-modal.tsx:12]: Should these be `TranslatedString`?

File: src/page-modules/contact/components/form/input.tsx [hunk 3]
```diff
@@ -4,13 +4,15 @@ import { PageText, TranslatedString, useTranslation } from '@atb/translations';
...
 import ErrorMessage from './error-message';
-import DescriptionModal from './description-modal';
 import { MonoIcon } from '@atb/components/icon';
 import { Button } from '@atb/components/button';
+import { DescriptionModal } from '.';
 
 type InputProps = {
   label: TranslatedString;
-  description?: string;
+  modalDescription?: string;
```
Comment (adriansberg) [inline src/page-modules/contact/components/form/input.tsx:13]: I'm wondering if the modal props should be an object (optional) and get its type from the `DescriptionModalProps`. I think that would make the input's code cleaner.

File: src/translations/pages/contact.ts [hunk 4]
```diff
@@ -1550,21 +1550,34 @@ export const Contact = {
...
+
+        _('På kvitteringen din', 'On your receipt ', 'På kvitteringa di'),
+        _(
+          'Du finner også ordre-id på utgåtte billetter',
+          'You can also find the order ID on expired tickets',
+          'Du finn også ordre-id på utgåtte billettar',
+        ),
+      ],
+      instruction: _(
+        `Gjelder forespørslen flere billetter, send ordre-id for alle billettene, separert med komma (' , ').`,
```
Comment (adriansberg) [inline src/translations/pages/contact.ts:1575]:
```suggestion
        `Gjelder forespørselen flere billetter, send ordre-id for alle billettene, separert med komma (' , ').`,
```

File: src/translations/pages/contact.ts [hunk 5]
```diff
@@ -1550,21 +1550,34 @@ export const Contact = {
...
+        `If the request applies to several tickets, send the order ID for all the tickets, separated by commas (' , ').`,
+        `Gjeld forespørslen flere billettar, send ordre-id for alle billettane, separert med komma (' , ').`,
```
Comment (adriansberg) [inline src/translations/pages/contact.ts:1577]:
```suggestion
        `Gjeld førespurnaden flere billettar, send ordre-id for alle billettane, separert med komma (' , ').`,
```

Maybe a different word has been used for this in other places? From what I can see "forespørsel" is not a valid nynorsk word.

**Review by adriansberg — CHANGES_REQUESTED** [review]
Added comments

File: src/page-modules/contact/components/form/form.module.css (hunk 1 above)
Comment (jonasbrunvoll) [reply src/page-modules/contact/components/form/form.module.css]: To be honest, I'm not sure why the wildcard is there in the first place. I remove it, as I see no difference whether it is there or not.

File: src/page-modules/contact/components/form/description-modal.tsx [hunk 6]
```diff
@@ -5,23 +5,28 @@ import { FocusScope } from '@react-aria/focus';
...
-import { PageText, useTranslation } from '@atb/translations';
+import { PageText, TranslatedString, useTranslation } from '@atb/translations';
 
 type DescriptionModalProps = {
   title: string;
-  description: string;
+  modalContent: {
+    description?: string;
+    instruction?: string;
+    bulletPoints?: TranslatedString[];
```
Comment (mortennordseth) [inline src/page-modules/contact/components/form/description-modal.tsx:15]: Why are description and instruction of type string, but bulletpoints not?

File: src/page-modules/contact/components/form/input.tsx [hunk 7]
```diff
@@ -22,12 +26,12 @@ export const Input = ({
...
   checked,
   value,
-  description,
+  modalContent,
   onChange,
 }: InputProps) => {
   const { t } = useTranslation();
   const [isModalOpen, setIsModalOpen] = useState(false);
-  const displayDescriptionModal = description && isModalOpen; // Condition to avoid rendering of useEffect inside DescriptionModal before opened.
+  const displayDescriptionModal = modalContent && isModalOpen; // Condition to avoid rendering of useEffect inside DescriptionModal before opened.
```
Comment (mortennordseth) [inline src/page-modules/contact/components/form/input.tsx:34]: Should you check if the modalContent actually has one of the properties set? None of the modalContent properties are currently required.

Comment (jonasbrunvoll) [reply src/page-modules/contact/components/form/input.tsx:34]: Right. Yes, at least on of the properties inside modal content should be present. Will update.

File: src/translations/pages/contact.ts [hunk 8]
```diff
@@ -1551,19 +1551,32 @@ export const Contact = {
...
-              "If you want help with a ticket you have already bought, we need to know the order ID. You can find it on the ticket in the app, or on your receipt. You can also find the order ID on expired tickets. If your request concerns several tickets, you must remember to send with the order ID for all the tickets. For multiple order IDs, separate with commas (',').",
-              "Viss du vil ha hjelp med ein billett du allereie har kjøpt, treng vi å vite ordre-id. Den finn du på billetten i appen, eller på kvitteringa di. Du finn ordre-id også på utgåtte billettar. Gjeld førespurnaden din fleire billettar, må du hugse å sende med ordre-id for alle billettane. Ved flere ordre-id-er, skill med komma (',').",
-            ),
 
+      description: _(
+        'Hvis du vil ha hjelp med en billett du allerede har kjøpt, trenger vi å vite ordre-id. Ordre-id finner du følgende steder:',
+        'If you want help with a ticket you have already bought, we need to know the order ID. You can find the order ID in the following places:',
+        'Viss du vil ha hjelp med ein billett du allereie har kjøpt, treng vi å vite ordre-id. Ordre-id finn du følgende steder:',
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:1558]:
```suggestion
        'Viss du vil ha hjelp med ein billett du allereie har kjøpt, treng vi å vite ordre-id. Ordre-id finn du følgende stadar:',
```

Comment (mortennordseth) [reply src/translations/pages/contact.ts:1558]: Evt. stader

**Review by mortennordseth — APPROVED** [review]

**Pull request merged**